var/spack/repos/builtin/packages/likwid/package.py

```diff
@@ -59,13 +56,18 @@ def filter_sbang(self):
                     *files)
 
     def install(self, spec, prefix):
-        if self.compiler.name not in self.supported_compilers:
+        supported_compilers = {'clang': 'CLANG', 'gcc': 'GCC', 'intel': 'ICC'}
+        if spec.target() == 'aarch64':
+            supported_compilers = {'gcc': 'GCCARMv8', 'clang': 'ARMCLANG'}
+        elif spec.target().startswith('ppc64'):
+            supported_compilers = {'gcc': 'GCCPOWER'}
+        if self.compiler.name not in supported_compilers:
```

Comment:
Can you add conflicts instead? That has two benefits:
1. It currently fails early, just after concretization, instead of failing after having built all the dependencies
2. When the new concretizer lands conflicts will be used for constraint resolution

Reply:
I'm not sure how to write the conflicts in that case. The documentation for `conflict` does not contain more complicated cases like:
`conflict("%clang" and (spec.target.family == "ppc" or spec.target.family == "ppc64"), "5.0.0:")`

Reply:
That would be:
```python
conflicts('%clang', when='@5.0.0: target=ppc64:')
```
Anyhow, I think it's good enough right now to merge the PR. We can move these statements to conflicts later.